pkg/util/azureclient/openshiftmanagedcluster/admin/client.go

```diff
@@ -667,3 +734,31 @@ func (future *OpenShiftManagedClustersRotateSecretsFuture) Result(client OpenShi
 	ar.Response = future.Response()
 	return
 }
+
+// OpenShiftManagedClustersClusterStatusFuture an abstraction for monitoring and retrieving the results of a
+// cluster status operation.
+type OpenShiftManagedClustersClusterStatusFuture struct {
+	azure.Future
+}
+
+func (future *OpenShiftManagedClustersClusterStatusFuture) Result(client OpenShiftManagedClustersClient) (ar autorest.Response, err error) {
+	var done bool
+	done, err = future.Done(client)
+	if err != nil {
+		err = autorest.NewErrorWithError(err, "containerservice.OpenShiftManagedClustersClusterStatusFuture", "Result", future.Response(), "Polling failure")
+		return
+	}
+	if !done {
+		err = azure.NewAsyncOpIncompleteError("containerservice.OpenShiftManagedClustersClusterStatusFuture")
+		return
+	}
+	ar.Response = future.Response()
+	return
+}
+
+// OpenShiftManagedClustersStatus contains the status of the control plane pods
+type OpenShiftManagedClustersStatus struct {
+	autorest.Response `json:"-"`
+
+	Items []v1.Pod `json:"pods,omitempty"`
```

Comment:
we absolutely do not want to introduce the kubernetes API definitions into our API or clients.